lnwallet/channel.go

```diff
@@ -2519,33 +2514,46 @@ func (lc *LightningChannel) CompleteCooperativeClose(remoteSig []byte) (*wire.Ms
 	closeTx := CreateCooperativeCloseTx(lc.fundingTxIn,
 		lc.channelState.OurDustLimit, lc.channelState.TheirDustLimit,
 		lc.channelState.OurBalance, lc.channelState.TheirBalance,
-		lc.channelState.OurDeliveryScript, lc.channelState.TheirDeliveryScript,
+		lc.channelState.OurDeliveryScript,
+		lc.channelState.TheirDeliveryScript,
 		lc.channelState.IsInitiator)
 
 	// Ensure that the transaction doesn't explicitly validate any
 	// consensus rules such as being too big, or having any value with a
 	// negative output.
 	tx := btcutil.NewTx(closeTx)
 	if err := blockchain.CheckTransactionSanity(tx); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
-	// With the transaction created, we can finally generate our half of
-	// the 2-of-2 multi-sig needed to redeem the funding output.
 	hashCache := txscript.NewTxSigHashes(closeTx)
-	lc.signDesc.SigHashes = hashCache
-	closeSig, err := lc.signer.SignOutputRaw(closeTx, lc.signDesc)
-	if err != nil {
-		return nil, err
+
+	// If we're the responder in the coop close workflow, our signature
+	// will have been generated in CreateCloseProposal and will be passed
+	// in as the localSig. If we're the initiator, we'll generate our
+	// half of the 2-of-2 multi-sig needed to redeem the funding output.
+	var ourSig []byte
+	var ourFlaggedSig []byte
```

Comment:
Namning suggestion: `ourFlaggedSig` -> `sigWithSighash`. 


Also this comment doesn't seem quite correct. Based on my interpretation, the _responder_ will be the one that calls this method w/o a signature in the case that they immediately accept the initiators proposal w.r.t the fee amount. The initiator will be the one that passes in a non-nil value for _both_ signatures as they currently hold the current pending signature in volatile memory.